src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -1045,97 +1075,207 @@ private static void ParseStatusLine(ReadOnlySpan<byte> line, HttpResponseMessage
             }
         }
 
-        private static void ParseHeaderNameValue(HttpConnection connection, ReadOnlySpan<byte> line, HttpResponseMessage response, bool isFromTrailer)
+        private bool ParseHeaders(HttpResponseMessage? response, bool isFromTrailer)
         {
-            Debug.Assert(line.Length > 0);
+            Span<byte> buffer = new Span<byte>(_readBuffer, _readOffset, _readLength - _readOffset);
+
+            (bool finished, int bytesConsumed) = ParseHeadersCore(buffer, response, isFromTrailer);
 
-            int pos = 0;
-            while (line[pos] != (byte)':' && line[pos] != (byte)' ')
+            int bytesScanned = finished ? bytesConsumed : buffer.Length;
+            if (_allowedReadLineBytes < bytesScanned)
```

Comment:
Why is this check `<` but the one at https://github.com/dotnet/runtime/pull/74393/files#diff-595f700a4c83b735d856fe14b5ebe8c7f7c18ae6e586adc8b398a997a0678274R997 which may also be comparing against buffer.Length uses `<=`?